interfaces/apparmor/spec.go

```diff
@@ -148,38 +148,41 @@ func (spec *Specification) AddLayout(si *snap.Info) {
 	// Append update-ns snippets that allow constructing the layout.
 	for _, path := range paths {
 		var buf bytes.Buffer
+		emit := func(f string, args ...interface{}) {
```

Comment:
Looking at the original PR and this now I'm slightly confused about what's the added value of this indirection, i.e. having EmitUpdateNSFunc that returns a simple function (that adds snippet to spec with de-duplication); why not simply change AddUpdateNS function to take variable number of args?

Reply:
The original PR has to split each of the buffers that were multi-line strings into actual strings. This means that each of the `fmt.Fprintf(&buf, ...)` line has to change because we don't want to just `strings.Split()` the contents of the buffer and hope for the best.

The indirection gives us a way to do that with one change, the change of the emit function, across the whole stack. Here in this PR the emit function prints to a buffer, just like before. In the original PR emit adds a single line snippet to the `snap-update-ns` profile.

Reply:
Ok, let me rephrase: why not simply have a public helper on spec:
`func Emit(f string, args ...interface{})`

+ make `GenWritable*` functions belong to Specification (no need to pass emit func around)?

Reply:
We could do that but that would impact the tests. Right now we have certain batches that are added. Tests measure those precisely.

If you look at the other PR, there is something like that there, it's actually `UpdateNSEmitFunc() func(f string, args ...interface{})`. The difference is that only the other PR changes the granularity, along with the huge impact to how tests look like.

Reply:
Allright, it's ok for this PR to keep it minimal, but I'll carry my arguments to the followup PR (sorry, don't mean to make your life harder, just trying to understand the reasoning :) )